ukvm/ukvm_cpu_aarch64.c

```diff
@@ -105,3 +105,15 @@ void aarch64_setup_memory_mapping(uint8_t *va_addr, uint64_t ram_size,
         phy_space_size -= map_size;
     }
 }
+
+void aarch64_mem_size(size_t *mem_size) {
+    size_t mem;
+    mem = (*mem_size / PMD_SIZE) * PMD_SIZE;
+    assert (mem <= *mem_size);
+    if (mem < *mem_size)
+        warnx("adjusting memory to %zu bytes", mem);
+    if (mem > AARCH64_MMIO_BASE)
+        err(1, "guest memory size %zu bytes exceeds the max size %ul bytes",
+            mem, AARCH64_MMIO_BASE);
```

Comment:
Here will get a compiler error because AARCH64_MMIO_BASE unsigned long, the the format excepts %ul.
I think this would be %ul?

Reply:
You're right about AARCH64_MMIO_BASE defined as `unsigned long`. Did you mean `%lu` for the format? `%ul` is not a legal format, perhaps that's a typo @hannesm?

Reply:
Ahh, Yes, it should be %lu :)